### ISSUE TITLE
How to determine why a container segfaults?

### ISSUE THREAD

**Issue by Sebas-** [dotnet/dotnet-docker#83]
I'm running a ASP.NET Core application in a docker container, and once in a while it segfaults (and because of restart always, restarts).

Since i'm quite interested to know why it segfaults, because it looks random, I want to know if there is a way to capture the core dump (if any)? No stack trace so it must be outside the managed code is my guess.

docker logs record only this:

`info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 0.6902ms 302`
`Segmentation fault (core dumped)`
`info: Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommandBuilderFactory[1]`

Which of course is far from useful :)

I've seen this with dotnet:1.0.0-core and preview2 onbuild images (only 2 I used for production testing), running a ASP.Net core 1.0.0 application.

**Comment by janvorli** [comment]
See https://github.com/docker/docker/issues/11740 for a discussion on getting core dumps in docker containers.

**Comment by Sebas-** [comment]
Thanks!

**Comment by MichaelSimons** [comment]
@Sebas- were you able to verify the suggestion?  If that works and your question is answered, please feel free to close this issue.  If not, let us know so we can work with you to resolve your question.

Thanks

**Comment by Sebas-** [comment]
Hi Michael,

Since it happens randomly I have not yet had a segfault yet since I've restart with the suggestions in the docker thread.

Since my question here is answered, I will close it.

Thanks for the support.

**Comment by Sebas-** [comment]
@janvorli 

I've closed this question, since I can now create core dumps, but since I now have a core dump, where should I ask a follow up question?

(the core dump seems somewhat useless since no debug symbols can be found)

Sebas

**Comment by janvorli** [comment]
@Sebas- you can get symbols here: https://dotnet.myget.org/gallery/dotnet-core. You will need symbols for Microsoft.NETCore.Runtime.CoreCLR, Microsoft.NETCore.Jit and Microsoft.NETCore.DotNetHost. So something like runtime.ubuntu.14.04-x64.Microsoft.NETCore.Runtime.CoreCLR (not sure which Linux is used by the docker).
When you click on "more..." next to the version of the package on the right of its name, you'll get to page where you can pick the version you want (it need to match the version of the coreclr you were running with). After you select the version, click on the "download symbols" and also to the "download" link.
After you get the packages, you can rename them to .zip and then open it. follow the subfolders in the "runtime" folder until you get to the leaf folder. You'll see *.so.dbg files in the symbols package and *.so and dotnet files in the non-symbols one. Copy them side by side with your core dump.
Then open the dump. I used to have some problems opening it in lldb, but in gdb, it worked fine:
```
gdb ./dotnet
core-file ./your_core_file
set solib-search-path .
```
Now you should be able to see the native part of the stack. GDB cannot show the managed stuff.

**Comment by Sebas-** [comment]
@janvorli 

Thanks for the info. I will see if I can figure out what versions are included in the docker image (microsoft/dotnet:1.0.0-core is debian 8, here's an idea, publish also a docker container with the suffix -debug that adds a new layer with the corresponding symbol files :) ).

When I have a some spare time today/tomorrow I will give it a try.

Sebas
